src/Views/FirmwareView.vala

```diff
@@ -166,6 +194,33 @@ public class About.FirmwareView : Gtk.Stack {
         update_list.show_all ();
     }
 
+    private void show_release (Gtk.ListBoxRow widget) {
+        if (widget is Widgets.FirmwareUpdateRow) {
+            var row = (Widgets.FirmwareUpdateRow) widget;
+            firmware_release_view.update_view (row.device, row.release);
+            deck.visible_child = firmware_release_view;
+
+            firmware_release_view.update.connect ((device, release) => {
+                on_update_start (device);
+
+                update.begin (device, release, (obj, res) => {
+                    update.end (res);
+                    on_update_end ();
+                });
+            });
```

Comment:
This should be connected when the firmware release view is constructed, otherwise you're creating a new signal connection every time the release view is shown